### PR TITLE
Detect language unambiguously, or raise explanatory exception

### PR DESCRIPTION
**Pull request by pjkundert**
Correctly and unambiguously determine Mnemonic language.

@prusnak, perhaps this is sufficiently clear that it doesn't make the reference implementation less tidy?

### CODE REVIEW

**Comment by prusnak** [comment]
Concept ACK

@matejcik What do you think about the approach?

**Comment by matejcik** [comment]
concept ACK but please fix the style check

File: src/mnemonic/mnemonic.py [hunk 1]
```diff
@@ -101,16 +101,16 @@ def normalize_string(txt: AnyStr) -> str:
...
-            mnemo = cls(lang)
-            if first in mnemo.wordlist:
-                return lang
-
-        raise ConfigurationError("Language not detected")
+        possible = set(cls(lang) for lang in cls.list_languages())
+        for word in code.split():
+            possible = set(p for p in possible if word in p.wordlist)
+            if not possible:
+                raise ConfigurationError(f"Language unrecognized for {word!r}")
```
Comment (moneymanolis) [inline src/mnemonic/mnemonic.py:110]: typo here `{word!r}`?

Comment (pjkundert) [reply src/mnemonic/mnemonic.py:110]: No, I don't think so -- the purpose is to reveal the source of the Exception -- that 'word' is not recognized in any known language.

**Comment by pjkundert** [comment]
OK, @matejcik, I've fixed the style_check issues (the remaining failures are unrelated to this pull)

**Comment by matejcik** [comment]
the style check is still failing for the text file

**Comment by matejcik** [comment]
ah but it's something that was in the repo already. I'll fix that.

**Pull request merged**